clients/pkg/promtail/targets/file/filetarget.go

```diff
@@ -159,6 +154,11 @@ func (t *FileTarget) run() {
 		close(t.done)
 	}()
 
+	err := t.sync()
```

Comment:
How is this avoiding the deadlock? I thought sync is locking as well.

Reply:
The caller of NewFileTarget is also the code that can take a lock (sync target), moving it to a goroutine will release the lock early.